mdantic/mdantic.py

```diff
@@ -74,38 +79,44 @@ def get_related_enum_helper(ty, visited, result):
                 get_related_enum_helper(sub_ty, visited, result)
 
 
-def mk_struct(cls, structs):
+def mk_struct(cls: BaseModel, structs: Dict[str, List[Field]]) -> None:
```

Comment:
it seems cls should be of type `type[BaseModel]` here, which would also fix the type checker warning later:
Argument of type "type[BaseModel]" cannot be assigned to parameter "cls" of type "BaseModel" in function ...

Reply:
Fixed to be type[BaseModel] instead. I think this fixed one of the other warnings as well